XWOpt/OptNode/Texture.cs

```diff
@@ -0,0 +1,126 @@
+ * without restriction, including without limitation the rights to use, copy, modify,
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to the following
+ * conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies
+ * or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE
+ * OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+using System;
+using System.Collections.ObjectModel;
+
+namespace SchmooTech.XWOpt.OptNode
+{
+    public class Texture : BaseNode
+    {
+        // Not sure what this number means.  Seems to be the same on most textures inside of the same OPT.
+        private int group;
+        private string name;
+        private int mipLevels = 0;
+
+        private byte[] texturePalletRefs;
+        private Collection<byte[]> mipPalletRefs = new Collection<byte[]>();
+
+        // 16 pallets at decreasing light levels.
+        // colors packed 5-6-5 blue, green, red.
+        private TexturePallet pallet = new TexturePallet();
+        private int width;
+        private int height;
+
+        public int Group { get => group; set => group = value; }
+        public string Name { get => name; set => name = value; }
+        public int Width { get => width; set => width = value; }
+        public int Height { get => height; set => height = value; }
+        public int MipLevels { get => mipLevels; set => mipLevels = value; }
+        public Collection<byte[]> MipPalletRefs { get => mipPalletRefs; }
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1814:PreferJaggedArraysOverMultidimensional", MessageId = "Member")]
+        [CLSCompliant(false)]
+        public TexturePallet Pallet { get => pallet; }
+
+        internal Texture(OptReader reader, int textureNameOffset) : base(reader)
+        {
+            // TODO: Check for alpha channel 0 26 node.
+            reader.ReadUnknownUseValue(0, this);
+
+            group = reader.ReadInt32();
+
+            // TODO: Detect pallet reuse.
+            var palletAddressOffset = reader.ReadInt32();
+
+            reader.Seek(textureNameOffset);
+            name = reader.ReadString(9);
+
+            reader.SeekShouldPointHere(palletAddressOffset, this);
+
+            // Skip pallet data for now to get pallet references.
+            var palletOffset = reader.ReadInt32();
+
+            reader.ReadUnknownUseValue(0, this);
```

Comment:
Found in `CORTN.OPT`, for `Tex00008` and `Tex00009`.

This value indicates where the palette data is stored.
When the value is 0, the palette offset points to the palette data.
Otherwise, the palette offset is ignored and the palette data is stored immediatly after the texture mip data.